### PR TITLE
Fix eslint warnings

### PR DESCRIPTION
**Pull request by IlyesBenAmara**
This PR adds necessary changes for suppressing most of eslint generated warnings

### CODE REVIEW

File: src/hooks/useAppConfig.tsx [hunk 1]
```diff
@@ -48,9 +48,10 @@ export const AppConfigProvider = ({
         return {
           ...field,
           formatValue: getFieldValueFormatter(field.fieldType),
-          isValid: (value: any) => isInputValid(value, field.rules)
+          isValid: (value: unknown) => isInputValid(value, field.rules)
         }
       })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (MatthewErispe) [inline src/hooks/useAppConfig.tsx:54]: @IlyesBenAmara can this comment be removed?

File: src/hooks/useAuth.tsx [hunk 2]
```diff
@@ -108,6 +108,7 @@ export const AuthProvider = ({ children }: AuthProviderProps): JSX.Element => {
       oauthRef.current = params
       validateOAuth(params)
     }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (MatthewErispe) [inline src/hooks/useAuth.tsx:111]: same as above

File: src/services/ApiClient.ts [hunk 3]
```diff
@@ -35,6 +35,23 @@ interface GoldenRecordResponse {
   expandedGoldenRecords: GoldenRecord[]
 }
 
+interface JempiGoldenRecord extends Pick<PatientRecord, 'sourceId' | 'uid'> {
```
Comment (MatthewErispe) [inline src/services/ApiClient.ts:38]: @IlyesBenAmara rename MpiGoldenRecord to ExpandedGoldenRecord and JempiGoldenRecord to GoldenRecord

File: src/services/ApiClient.ts [hunk 4]
```diff
@@ -35,6 +35,23 @@ interface GoldenRecordResponse {
...
+interface MpiGoldenRecord {
+  goldenRecord: JempiGoldenRecord
+  mpiPatientRecords: Array<MpiPatientRecord>
+}
+
+interface MpiPatientRecord {
+  patientRecord: JempiPatientRecord
+}
+
+interface JempiPatientRecord extends Pick<PatientRecord, 'sourceId' | 'uid'> {
```
Comment (MatthewErispe) [inline src/services/ApiClient.ts:51]: @IlyesBenAmara apply similar changes as above for patient

**Review by MatthewErispe — CHANGES_REQUESTED** [review]

File: src/hooks/useAppConfig.tsx (hunk 1 above)
Comment (IlyesBenAmara) [reply src/hooks/useAppConfig.tsx:54]: If we remove that comment, the warning will show when we are linting the code

**Review by MatthewErispe — APPROVED** [review]

**Pull request merged**